waterbox/readme.txt

```diff
@@ -9,25 +9,27 @@ It consists of a modified musl libc, and build scripts to tie it all together.
 	PREPARE A WIN10 WORKSTATION:
 	Using the guidance at (https://docs.microsoft.com/en-us/windows/wsl/wsl2-kernel & https://docs.microsoft.com/en-us/windows/wsl/install-win10):
 	1. Install WSL2 
-	2. Install Ubuntu 20.04 LTS (https://www.microsoft.com/en-us/p/ubuntu-2004-lts/9n6svws3rx71)
-	3. Clone the bizhawk repository. You can use it through /mnt or /home if you really like 
-	4. Install build tools: sudo apt-get update && sudo apt-get install gcc g++ make cmake llvm zstd
-	4b. (Note for future work: ideally the llvm installed above would not be required)
+	2. Install Ubuntu 22.04.1 LTS (https://apps.microsoft.com/store/detail/ubuntu-22041-lts/9PN20MSR04DW)
+	3. Clone the bizhawk repository. You can use it through /mnt or /home if you really like
+	4. The waterbox toolchain has a choice between gcc and clang. Currently, clang is the preferred compiler due to its generally superior performance all around.
+	4a. (Clang) Install build tools: sudo apt-get update && sudo apt-get install make cmake clang lld llvm zstd
+	4b. (GCC) Install build tools: sudo apt-get update && sudo apt-get install make cmake gcc-12 g++-12 llvm zstd
+	5. Currently, GCC 12 is required when using a gcc environment (older gcc versions fail at building libcxx), this may change in the future.
 
 	PREPARE A WIN10 VM:
 	1. Make sure the VM has "yo dawg" virtualization enabled on the guest. For example in VMWare Workstation, "Virtualize Intel VT-X/EPT or AMD-V/RVI"
 	2. Follow WIN10 Workstation preparation guide
 	3. If you wish to clone bizhawk on your host system (slower but maybe more convenient), you can use a VMWare shared folder and: `sudo mount -t drvfs Z: /mnt/z -o rw,relatime,metadata,uid=1000,gid=1000` (WSL won't auto-mount the shared drive)
 
 	PREPARE A LINUX WORKSTATION:
-	1. TODO. This should work, but no one has tested it yet
+	1. Debian/Ubuntu based distros work fine with the WIN10 Workstation instructions (confirmed working with Debian 11). Other distros should be able provide the needed programs in their package managers. make, cmake, gcc/g++ or clang/clang++, ld or lld (lld is required for clang), gcc-ar/gcc-ranlib or llvm-ar/llv-ranlib (which one you choose doesn't matter), llvm-config, zstd.
 
 2. Clone bizhawk sources
 	* Make sure git's core.autocrlf is set to false, as the alternatives cause git to modify the line endings in .sh-looking files which WILL break the build process, from the very first step.
 	* This is NOT git's default. You will need to change it!! Go ahead and set it false globally permanently, since do you really want git modifying files?
 	* Make sure you have initialized and updated the needed submodules in the waterbox directory (for example, /waterbox/llvm-project and /waterbox/musl, etc.)

```

Comment:
There should be a note here that the `llvm-project` submodule can be checked out faster by using the setup-llvm script.
 Similarly the `libcxx/readme.txt` should be updated.